src/components/common/LocationMarker/index.tsx

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react';
+import { Marker, useMap } from 'react-leaflet';
+import customLocationIcon from '../../../constants/customLocationIcon';
+
+function LocationMarker() {
+  const map = useMap();
+  const [position, setPosition] = useState({ lat: 0, lng: 0 });
+
+  useEffect(() => {
+    map.locate().on('locationfound', (e) => {
+      setPosition(e.latlng);
+      map.flyTo(e.latlng, map.getZoom(), { animate: false });
+    });
+  }, []);
+
+  if (!position) return null;
+
+  return <Marker position={position} icon={customLocationIcon} />;
```

Comment:
WDYT change that for an [Inline If with Logical && Operator](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator
)

Reply:
Done!